omf-bifrost/src/admin/mod.rs

```diff
@@ -0,0 +1,381 @@
+        let pedestrian = access_rules
+            .get(&HighwayType::Pedestrian)
+            .and_then(|modes| modes_to_bitmask(modes));
+        let bridleway = access_rules
+            .get(&HighwayType::Bridleway)
+            .and_then(|modes| modes_to_bitmask(modes));
+        let cycleway = access_rules
+            .get(&HighwayType::Cycleway)
+            .and_then(|modes| modes_to_bitmask(modes));
+        let path = access_rules
+            .get(&HighwayType::Path)
+            .and_then(|modes| modes_to_bitmask(modes));
+        let motorroad = access_rules
+            .get(&HighwayType::Motorroad)
+            .and_then(|modes| modes_to_bitmask(modes));
+
+        let params: [&dyn rusqlite::ToSql; 10] = [
+            &trunk,
+            &trunk_link,
+            &track,
+            &footway,
+            &pedestrian,
+            &bridleway,
+            &cycleway,
+            &path,
+            &motorroad,
+            country_code as &dyn rusqlite::ToSql,
+        ];
+        let updated = stmt.execute(rusqlite::params_from_iter(params))?;
+        if updated == 0 {
+            log::warn!(
+                "No matching admin row found for admin_access rule '{}'. Ignore if not using a planet extract",
+                country_code
+            );
+        }
+    }
+
+    Ok(())
+}
+
+/// Build the admin and access SQLite database from Overture DuckDB-derived geo-parquet division and area files, using a given AdminConfig.
+/// The Spatialite extension must be available in the environment for spatial support.
+pub fn build_admins_from_geo_parquet(
+    geoparquet_division_path: &str,
+    geoparquet_area_path: &str,
+    sqlite_path: &str,
+    admin_config: &AdminConfig,
+) -> Result<()> {
+    info!(
+        "Preparing to build admins from {} and {} into {}",
+        geoparquet_division_path, geoparquet_area_path, sqlite_path
+    );
+
+    if Path::new(sqlite_path).exists() {
+        info!("Removing existing SQLite file at {}", sqlite_path);
+        fs::remove_file(sqlite_path)?;
+    }
+    let sqlite_con = Connection::open(sqlite_path)?;
+
+    // Set SQLite pragmas for faster bulk import (less durability during import but higher performance)
+    sqlite_con.execute_batch(
+        "PRAGMA synchronous = OFF;
+             PRAGMA journal_mode = MEMORY;
+             PRAGMA temp_store = MEMORY;",
+    )?;
+
+    unsafe {
+        sqlite_con.load_extension_enable()?;
+        sqlite_con.load_extension("mod_spatialite", None::<&str>)
+            .context("Failed to load mod_spatialite extension. Make sure SpatiaLite is installed and 'mod_spatialite' is available in your library path.")?;
+        sqlite_con.load_extension_disable()?;
+    }
+    sqlite_con.execute_batch("SELECT InitSpatialMetaData(1);")?;
+
+    let duck_con = DuckConnection::open_in_memory()?;
+    info!("Installing and loading DuckDB spatial extension");
+    duck_con.execute("INSTALL spatial", [])?;
+    duck_con.execute("LOAD spatial", [])?;
```

Comment:
Not needed for this PR, but just noting it would probably be good for us to wrap this and maybe the sqlite setup in a utility function as some point. `open_duckdb_spatial_conn()` or whatever